oe_multilingual.module

```diff
@@ -81,3 +83,77 @@ function oe_multilingual_configurable_language_presave(EntityInterface $entity)
     }
   }
 }
+
+/**
+ * Implements hook_block_view_BASE_BLOCK_ID_alter() for the Page Header block.
+ */
+function oe_multilingual_block_view_oe_theme_helper_page_header_alter(array &$build, BlockPluginInterface $block) {
+  $build['#pre_render'][] = 'oe_multilingual_page_header_pre_render';
+}
+
+/**
+ * Pre-render callback for the Page Header block alteration.
+ *
+ * We use this to add the language switcher
+ * to the page header if OpenEuropa Theme is being used.
+ *
+ * @param array $build
+ *   The built render array of the block.
+ *
+ * @see \Drupal\oe_theme_helper\Plugin\Block\PageHeaderBlock
+ *
+ * @return array
+ *   The built render array of the block.
+ */
+function oe_multilingual_page_header_pre_render(array $build): array {
+  // Get required services.
+  $multilingual_helper = \Drupal::service('oe_multilingual.helper');
+  $content_language_switcher_provider = \Drupal::service('oe_multilingual.content_language_switcher_provider');
+  $language_manager = \Drupal::languageManager();
+
+  $entity = $multilingual_helper->getEntityFromCurrentRoute();
+  // Bail out if there is no entity or if it's not a content entity.
+  if (!$entity || !$entity instanceof ContentEntityInterface) {
+    return $build;
+  }
+
+  // Render the links only if the current entity translation language is not
+  // the same as the current site language.
+  /** @var \Drupal\Core\Entity\EntityInterface $translation */
+  $translation = $multilingual_helper->getCurrentLanguageEntityTranslation($entity);
+  $current_language = $language_manager->getCurrentLanguage();
+  if ($translation->language()->getId() === $current_language->getId()) {
+    return $build;
+  }
+
+  $content = &$build['content'];
+
+  $content['#language_switcher']['current'] = $translation->language()->getName();
+
+  /** @var \Drupal\Core\Language\LanguageInterface[] $languages */
+  $languages = $language_manager->getNativeLanguages();
+  $content['#language_switcher']['unavailable'] = $languages[$current_language->getId()]->getName();
+
+  // Normalize the links to an array of options suitable for the ECL
+  // "ecl-lang-select-pages" template.
+  $content['#language_switcher']['options'] = [];
+  foreach ($content_language_switcher_provider->getAvailableEntityLanguages($entity) as $language_code => $link) {
+    /** @var \Drupal\Core\Url $url */
+    $url = $link['url'];
+    $href = $url
+      ->setOptions(['language' => $link['language']])
+      ->setAbsolute(TRUE)
+      ->toString();
+
+    $content['#language_switcher']['options'][] = [
+      'href' => $href,
+      'hreflang' => $language_code,
+      'label' => $link['title'],
+      'lang' => $language_code,
+    ];
+  }
+
+  $content['#language_switcher']['is_primary'] = TRUE;
+
+  return $build;
```

Comment:
Shouldn't we add the entity as cachable dependency?

Reply:
According to @upchuk we shouldn't, according to this: https://www.drupal.org/node/2232375

Reply:
Yupp.